File: micrometer-core/src/test/groovy/io/micronaut/docs/MethodNameTagger.java
package io.micronaut.docs;

import io.micrometer.core.instrument.Tag;
import io.micronaut.aop.MethodInvocationContext;
import io.micronaut.configuration.metrics.aggregator.AbstractMethodTagger;
import jakarta.inject.Singleton;

import java.util.Collections;
import java.util.List;

@Singleton
public class MethodNameTagger extends AbstractMethodTagger {
    @Override
    public List<Tag> buildTags(MethodInvocationContext<Object, Object> context) {
        return Collections.singletonList(Tag.of("method", context.getMethodName()));
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/metrics/annotation/MethodNameTagger.kt
package io.micronaut.docs.metrics.annotation

// tag::imports[]
import io.micrometer.core.instrument.Tag
import io.micronaut.aop.MethodInvocationContext
import io.micronaut.configuration.metrics.aggregator.AbstractMethodTagger
import jakarta.inject.Singleton
// end::imports[]

// tag::class[]
@Singleton
class MethodNameTagger : AbstractMethodTagger() {
    override fun buildTags(context: MethodInvocationContext<Any, Any>): List<Tag> =
        listOf(Tag.of("method", context.methodName))
}
// end::class[]
